### PR TITLE
[Cosmos] [Preview] Full-Fidelity Change Feed

### PR DESCRIPTION
**Pull request by simorenoh**
This PR adds functionality needed to use the full-fidelity change feed with Change Feed Processor and Change Feed Pull Model. 

Reference PR for some of the wire format changes: https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3197

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/HttpConstants.java [hunk 1]
```diff
@@ -323,6 +324,10 @@ private static String getSdkVersionWithSnapshotInsteadOfBeta() {
         }
     }
 
+    public static class ChangeFeedWireFormatVersions {
```
Comment (milismsft) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/HttpConstants.java:327]: why do we need to capture what looks like an API version here?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/fullfidelity/ChangeFeedMetadata.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+package com.azure.cosmos.implementation.changefeed.fullfidelity;
+
+import com.azure.cosmos.util.Beta;
+
+import java.time.Instant;
+
+/**
+ * The metadata of a change feed resource with ChangeFeedMode set to FULL_FIDELITY
+ */
+@Beta(value = Beta.SinceVersion.V4_28_0)
```
Comment (milismsft) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/fullfidelity/ChangeFeedMetadata.java:12]: I believe we are way passed the 4.28...

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/fullfidelity/ChangeFeedOperationType.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.changefeed.fullfidelity;
+
+import com.azure.cosmos.util.Beta;
+
+@Beta(value = Beta.SinceVersion.V4_28_0)
+public enum ChangeFeedOperationType {
```
[the remark this answers is not in the document]
Comment (simorenoh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/fullfidelity/ChangeFeedOperationType.java:8]: these were the only operation types that were in the .NET PR

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/HttpConstants.java (hunk 1 above)
Comment (simorenoh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/HttpConstants.java:327]: it's not needed but it is an improvement in the content that is returned - https://msdata.visualstudio.com/CosmosDB/_wiki/wikis/CosmosDB.wiki/34015/dotNET

Comment (milismsft) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/HttpConstants.java:327]: This should be an "enum" class then, if we ever plan to support other CF wire formats.

**Review by milismsft — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/fullfidelity/ChangeFeedOperationType.java (hunk 3 above)
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/fullfidelity/ChangeFeedOperationType.java:8]: Same here. this is implementation. We don't need Beta annotations here. 
Also, are we planning to ship `ChangeFeedOperationType` and `ChangeFeedMetaData` as part of `handle()` API response?

**Pull request closed without merging**